dotnet/src/Connectors/Connectors.GoogleVertexAI/Settings/GeminiPromptExecutionSettings.cs

```diff
@@ -25,46 +35,135 @@ public sealed class GeminiPromptExecutionSettings : PromptExecutionSettings
     /// Range is 0.0 to 1.0.
     /// </summary>
     [JsonPropertyName("temperature")]
-    public double? Temperature { get; set; }
+    public double? Temperature
+    {
+        get => this._temperature;
+        set
+        {
+            this.ThrowIfFrozen();
+            this._temperature = value;
+        }
+    }
 
     /// <summary>
     /// TopP controls the diversity of the completion.
     /// The higher the TopP, the more diverse the completion.
     /// </summary>
     [JsonPropertyName("top_p")]
-    public double? TopP { get; set; }
+    public double? TopP
+    {
+        get => this._topP;
+        set
+        {
+            this.ThrowIfFrozen();
+            this._topP = value;
+        }
+    }
 
     /// <summary>
     /// Gets or sets the value of the TopK property.
     /// The TopK property represents the maximum value of a collection or dataset.
     /// </summary>
     [JsonPropertyName("top_k")]
-    public int? TopK { get; set; }
+    public int? TopK
+    {
+        get => this._topK;
+        set
+        {
+            this.ThrowIfFrozen();
+            this._topK = value;
+        }
+    }
 
     /// <summary>
     /// The maximum number of tokens to generate in the completion.
     /// </summary>
     [JsonPropertyName("max_tokens")]
-    public int? MaxTokens { get; set; }
+    public int? MaxTokens
+    {
+        get => this._maxTokens;
+        set
+        {
+            this.ThrowIfFrozen();
+            this._maxTokens = value;
+        }
+    }
 
     /// <summary>
     /// The count of candidates. Possible values range from 1 to 8.
     /// </summary>
     [JsonPropertyName("candidate_count")]
-    public int? CandidateCount { get; set; }
+    public int? CandidateCount
+    {
+        get => this._candidateCount;
+        set
+        {
+            this.ThrowIfFrozen();
+            this._candidateCount = value;
+        }
+    }
 
     /// <summary>
     /// Sequences where the completion will stop generating further tokens.
     /// Maximum number of stop sequences is 5.
     /// </summary>
     [JsonPropertyName("stop_sequences")]
-    public IList<string>? StopSequences { get; set; }
+    public IList<string>? StopSequences
+    {
+        get => this._stopSequences;
+        set
+        {
+            this.ThrowIfFrozen();
+            this._stopSequences = value;
+        }
+    }
 
     /// <summary>
     /// Represents a list of safety settings.
     /// </summary>
     [JsonPropertyName("safety_settings")]
-    public IList<GeminiSafetySetting>? SafetySettings { get; set; }
+    public IList<GeminiSafetySetting>? SafetySettings
+    {
+        get => this._safetySettings;
+        set
+        {
+            this.ThrowIfFrozen();
+            this._safetySettings = value;
+        }
+    }
+
+    /// <inheritdoc />
+    public override void Freeze()
```

Comment:
Potential bug fix for our `PromptExecutionSettings`. 

Don't freeze if it's already frozen, ignore and return right away.

Reply:
Done. I also created PR for main branch https://github.com/microsoft/semantic-kernel/pull/5057